fvm/meter/weighted/meter.go

```diff
@@ -75,6 +196,16 @@ func WithComputationWeights(weights ExecutionEffortWeights) WeightedMeterOptions
 
 // WithMemoryWeights sets the weights for the memory intensities
 func WithMemoryWeights(weights ExecutionMemoryWeights) WeightedMeterOptions {
+	// Replace some weights
+	if int(common.MemoryKindLast)-len(weights) != 1 {
```

Comment:
Might be simpler to compare with `DefaultMemoryWeights`?
i.e:
```suggestion
	if len(weights) != len(DefaultMemoryWeights) {
```